### ISSUE TITLE
Random execution order of test cases (shuffle)

### ISSUE THREAD

**Issue by shreyasbharath** [ThrowTheSwitch/Unity#2]
This is a very useful feature that I've come across in GoogleTest (using the --gtest-shuffle option). The main reason you'd want to shuffle tests is to expose any tests that depend on any other test in any way.

I can't use GoogleTest on the embedded target I am working on because it is too heavy weight. Would this be a difficult feature to add?

I'll try toying with an implementation on my fork and submit a pull request when it is ready.

Thanks.

**Comment by jamestiotio** [comment]
Hi folks, I stumbled upon this issue while trying to check whether Unity supports this shuffled random execution order of tests or not. This issue was [closed as completed](https://github.com/ThrowTheSwitch/Unity/issues/2#event-100067582) by @mvandervoord, but as far as I am aware, I do not think Unity supports this feature yet, and there are no corresponding relevant pull requests or commits linked to this issue that introduced this feature into Unity.

Can I confirm whether this feature has been added to Unity or not? If yes, can I ask how to activate it (and perhaps add those steps to the docs)? If not, then I am willing to raise a PR to add this feature, including adding the aforementioned possibility of using a seed for deterministic reproduction.

Thanks!
